### PR TITLE
session_test: move some tests.

### PR DESCRIPTION
**Pull request by hanfei1991**
1. move InTrans test
2. move Prepare Retry test
3. move Auth Test
4. Remove Substring_index test, which is in expression's test.

### CODE REVIEW

**Comment by hanfei1991** [comment]
@coocood @tiancaiamao PTAL

File: session_test.go [hunk 1]
```diff
@@ -1458,66 +1377,6 @@ func (s *test1435Suite) TestIssue1435(c *C) {
...
-	c.Assert(se.Auth(&auth.UserIdentity{Username: "xxx", Hostname: "%"}, []byte("yyy"), []byte("zzz")), IsTrue)
-	c.Assert(se.Auth(&auth.UserIdentity{Username: "root", Hostname: "%"}, []byte(""), []byte("")), IsTrue)
-	mustExecSQL(c, se, "create table t (id int)")
-
-	privileges.Enable = save1
-	privileges.SkipWithGrant = save2
-	mustExecSQL(c, se, dropDBSQL)
-}
-
-func (s *testSessionSuite) TestSubstringIndexExpr(c *C) {
```
Comment (shenli) [inline session_test.go]: This test case is discarded?

Comment (hanfei1991) [reply session_test.go]: yes, the description said "Remove Substring_index test, which is in expression's test."

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**